Lib/logging/handlers.py

```diff
@@ -582,17 +582,24 @@ def send(self, s):
         This function allows for partial sends which can happen when the
         network is busy.
         """
-        if self.sock is None:
-            self.createSocket()
-        #self.sock can be None either because we haven't reached the retry
-        #time yet, or because we have reached the retry time and retried,
-        #but are still unable to connect.
-        if self.sock:
+        attempt = True
+        while True:
+            if self.sock is None:
+                attempt = False
+                self.createSocket()
+                # self.sock can be None either because we haven't reached the retry
+                # time yet, or because we have reached the retry time and retried,
+                # but are still unable to connect.
+                if self.sock is None:
+                    break
             try:
                 self.sock.sendall(s)
             except OSError: #pragma: no cover
                 self.sock.close()
                 self.sock = None  # so we can call createSocket next time
+                if attempt:
+                    continue
+            break
```

Comment:
IIUC, this code will retry connection once, straight away, if the handler was previously connected but an error occurs when actually sending the event. In the case where the reason for the error is that the remote has gone offline, how likely is that reconnection attempt to succeed? Isn't the remote going offline for a period the case described in the issue?